src/Lean/Server/Rpc/Basic.lean

```diff
@@ -33,36 +40,98 @@ end Lean.Lsp
 
 namespace Lean.Server
 
-structure RpcObjectStore : Type where
-  /-- Objects that are being kept alive for the RPC client, together with their type names,
-  mapped to by their RPC reference.
+/--
+Marks values to be encoded as opaque references in RPC packets.
+Two `WithRpcRef`s with the same `id` will yield the same `RpcRef`.
```

Comment:
```suggestion
Two `WithRpcRef`s with the same `id` will yield the same client-side reference.
```
Maybe we can treat `RpcRef` as an impl detail and avoid reference in user-facing docs.